core/storage/question/gae_models.py

```diff
@@ -132,3 +169,83 @@ def create(
             question_id=question_id, skill_id=skill_id)
 
         return question_skill_link_model_instance
+
+
+class QuestionCommitLogEntryModel(base_models.BaseCommitLogEntryModel):
+    """Log of commits to questions.
+
+    A new instance of this model is created and saved every time a commit to
+    QuestionModel occurs.
+
+    The id for this model is of the form
+    'question-{{QUESTION_ID}}-{{QUESTION_VERSION}}'.
+    """
+    # The id of the question being edited.
+    question_id = ndb.StringProperty(indexed=True, required=True)
+
+    @classmethod
+    def _get_instance_id(cls, question_id, question_version):
+        """Returns ID of the question commit log entry model.
+
+        Args:
+            question_id: str. The question id whose states are mapped.
+            question_version: int. The version of the question.
+
+        Returns:
+            str. A string containing question ID and
+                question version.
+        """
+        return 'question-%s-%s' % (question_id, question_version)
+
+
+class QuestionSummaryModel(base_models.BaseModel):
+    """Summary model for an Oppia question.
+
+    This should be used whenever the content blob of the question is not
+    needed (e.g. in search results, etc).
+
+    A QuestionSummaryModel instance stores the following information:
+
+    creator_id, language_code, status, question_model_last_updated,
+    question_model_created_on, question_data.
+
+    The key of each instance is the question id.
+    """
+    # The user ID of the creator of the question.
+    creator_id = ndb.StringProperty(required=True)
+    # The ISO 639-1 code for the language this question is written in.
+    language_code = ndb.StringProperty(required=True, indexed=True)
+    # The status of the question. It can be 'private', 'approved',
+    # 'rejected' and 'pending'.
+    status = ndb.StringProperty(
```

Comment:
Interesting choice to put this in the summary. In general, though, the summary is supposed to be derivable from QuestionModel.

I think we should think of this like suggestions. Have a QuestionSuggestionModel that basically contains a "question suggestion". Only make it into a QuestionModel once it's approved.

But perhaps @nithusha21 could advise here. /cc @anmolshkl too.

Reply:
Moved it to QuestionModel.